xicam/plugins/__init__.py

```diff
@@ -63,19 +62,40 @@ class Filters(Enum):
     COMPLETE = auto()
 
 
+class Status(Enum):
+    LoadingQueue = auto()
+    Loading = auto()
+    InstantiateQueue = auto()
+    Instantiating = auto()
+    Success = auto()
+    FailedLoad = auto()
+    FailedInstantiate = auto()
+
+
+@dataclass
```

Comment:
this is interesting, so this decorator gives us some special dunder methods for free (like init and repr)? Just to learn, why did you do this here instead of implementing the class special methods yourself?

Reply:
My though process was something like:
- I'll just make a custom class for each type of task
- If I use a namedtuple with defaults, its more concise and unified
- ...but I want it to be mutable so I can update `.state`, so I try using a `dataclass`
- Wow, `dataclass`'s are great, I can even tune how equality is evaluated!

I could implement all that on my own in custom class, but why bother, right?